### PR TITLE
mgr/cephadm is not defining haproxy tcp healthchecks for Ganesha

### PR DESCRIPTION
**Pull request by avanthakkar**
Fixes: https://tracker.ceph.com/issues/62638

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by phlogistonjohn — COMMENTED** [review]
Syntax looks OK to me.  What's the goal of `weight 100`? I looked at the docs and it wasn't clear to me what the value of setting it for nfs is here.

What's the verification plan for the change? I'm guessing you're testing it manually for now? If so, how?  Did you look into writing a tuethology test for this, was it too complex to implement?

**Comment by avanthakkar** [comment]
> Syntax looks OK to me. What's the goal of `weight 100`? I looked at the docs and it wasn't clear to me what the value of setting it for nfs is here.
> 
> What's the verification plan for the change? I'm guessing you're testing it manually for now? If so, how? Did you look into writing a tuethology test for this, was it too complex to implement?

As per our discussion yesterday in standup does it makes sense to leave out the `weight 100` change for now & just address the healthcheck, keeping scope accurate to the tracker issue mentioned? Also explore the weight load among nfs servers and see how it works out behind HA. Wdyt? @phlogistonjohn  @adk3798

**Comment by adk3798** [comment]
> > Syntax looks OK to me. What's the goal of `weight 100`? I looked at the docs and it wasn't clear to me what the value of setting it for nfs is here.
> > What's the verification plan for the change? I'm guessing you're testing it manually for now? If so, how? Did you look into writing a tuethology test for this, was it too complex to implement?
> 
> As per our discussion yesterday in standup does it makes sense to leave out the `weight 100` change for now & just address the healthcheck, keeping scope accurate to the tracker issue mentioned? Also explore the weight load among nfs servers and see how it works out behind HA. Wdyt? @phlogistonjohn @adk3798

all sounds right to me. The health check itself should in theory be safe and we can spend more time investigating the effects of the weights.

File: src/pybind/mgr/cephadm/templates/services/ingress/haproxy.cfg.j2
Comment (adk3798) [inline src/pybind/mgr/cephadm/templates/services/ingress/haproxy.cfg.j2:88]: There are 3 unit tests
```
FAILED cephadm/tests/test_services.py::TestIngressService::test_ingress_config_nfs_multiple_nfs_same_rank[False]
FAILED cephadm/tests/test_services.py::TestIngressService::test_ingress_config_nfs_multiple_nfs_same_rank[True]
FAILED cephadm/tests/test_services.py::TestIngressService::test_ingress_config_nfs_proxy_protocol
```
that generate haproxy configs that will have to be updated for make check to pass with this change.

Comment (avanthakkar) [reply src/pybind/mgr/cephadm/templates/services/ingress/haproxy.cfg.j2]: Thanks. Fixed the tests!

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-10-12_12:04:04-orch:cephadm-wip-adk-testing-2023-10-11-2128-distro-default-smithi/

1 failure: https://tracker.ceph.com/issues/62958, known issue

Nothing to block merging

**Comment by avanthakkar** [comment]
> https://pulpito.ceph.com/adking-2023-10-12_12:04:04-orch:cephadm-wip-adk-testing-2023-10-11-2128-distro-default-smithi/
> 
> 1 failure: https://tracker.ceph.com/issues/62958, known issue
> 
> Nothing to block merging

Can we have this merged once CI passes? @adk3798

**Comment by avanthakkar** [comment]
jenkins test api

**Comment by adk3798** [comment]
jenkins test api

**Review by adk3798 — APPROVED** [review]
tentatively approving with the weight portion of this removed, as discussed in the standup meeting.

**Comment by avanthakkar** [comment]
jenkins test api

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-02-27_05:08:20-orch:cephadm-wip-adk-testing-2024-02-26-1600-distro-default-smithi/

not going to go over the failures in the regular detail as we're still cleaning up the `in cluster log` stuff, but it was mostly just things we still need to add to ignorelists and the mds_upgrade_sequence test having issues. Only test that failed due to PRs in the run was https://pulpito.ceph.com/adking-2024-02-27_05:08:20-orch:cephadm-wip-adk-testing-2024-02-26-1600-distro-default-smithi/7574464 so we can't merge the PRs making changes to mgr/nfs but everything else seems to be okay.

**Pull request merged**
